foundry/main.rs

```diff
@@ -60,16 +61,47 @@ fn run() -> Result<(), String> {
         .about("Foundry client")
         .subcommand(SubCommand::with_name("commit-hash").about("Print the commit hash of the source tree"));
 
+    let app = app.arg(
+        clap::Arg::with_name("module_arguments")
+            .short("D")
+            .help("Override config values in app-descriptor. For example  '-D foo=bar -D x=y'")
+            .takes_value(true)
+            .multiple(true),
+    );
+
     let matches = Config::parse_args(app);
     let config_path = matches.value_of("config").unwrap_or("./config.ini");
     let conf = Config::parse_file(&matches, config_path).expect("read config file");
 
+    let module_arguments = parse_module_args(matches.values_of("module_arguments"));
+
     match matches.subcommand_name() {
         Some(_) => run_subcommand(&matches),
-        None => run_node(conf, None),
+        None => run_node(conf, module_arguments, None),
     }
 }
 
+fn parse_module_args(args: Option<Values>) -> BTreeMap<String, String> {
+    let mut ret = BTreeMap::new();
+
+    if args.is_none() {
+        return ret
+    }
+
+    let vals: Vec<&str> = args.unwrap().collect();
+    for keyvalue in vals {
+        let splitted: Vec<&str> = keyvalue.split('=').collect();
+        if splitted.len() != 2 {
+            panic!("Invalid module arguments. Please use -D key=value format");
```

Comment:
Is it typical to panic on wrong argument?

Reply:
https://github.com/clap-rs/clap/blob/master/examples/15_custom_validator.rs

This seems a more legitimate way to deal with a custom value format.

Reply:
It uses only validation.
How about using `values_t!(...).unwrap_or_else(|e| e.exit())`?
`exit` is more user friendly since it does not print unnecessary call stacks.
https://github.com/clap-rs/clap/blob/v2-master/examples/12_typed_values.rs#L48

Reply:
Ok. I thought that the validation failure should trigger printing of proper error messages.